tasks/build.py

```diff
@@ -135,6 +135,38 @@ def create_directory(pr, jobs_base_dir, event_info):
     return ym, pr_id, run_dir
 
 
+def run_cmd(run_command, cmd_detail, arch_job_dir):
+    """Run Commands and logs the process
+
+    Args:
+        run_command (string): command to run 
+        cmd_detail (string): purpose of the commant
+        arch_job_dir (string): location of arch_job_dir
+
+    Returns:   
+        tuple of  2 elements containing
+
+        - process_output(string): output of the process
+        - process_exit_code(string): exit code of the process   
+    """
+    log(" '%s' by running '%s' in directory '%s'" % (cmd_detail, run_command, arch_job_dir))
+    result = subprocess.run(run_command,
+                                cwd=arch_job_dir,
+                                shell=True,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+    process_output = result.stdout.decode()
+    process_error = result.stderr.decode()
+    process_exit_code = result.returncode
+
+    if process_exit_code != 0:
+        log("Error: '%s' Exitcode: %s" % (process_error,process_exit_code))
```

Comment:
We should always log the exit code, even if it's zero.

Optionally raise an exception if exit code is non-zero:

```python
def run_cmd(cmd, log_msg='', parent_dir=None, raise_error=True):

    log(" '%s' \nexit code: %s\nStdout %s\nStderr: %s" % (cmd_detail, process_exit_code, result.stdout,result.stderr))
    if process_exit_code != 0 and raise_error:
        raise RuntimeError("...." % (cmd, process_exit_code))

Reply:
While raising exceptions may be good in general, it may complicate the refactoring. Would favour to skip this here. Otherwise we would have to think about when to raise exceptions, when not, etc.

Reply:
OK, let deal with that in issue #6